agent-codeless/src/test/java/com/microsoft/applicationinsights/agentc/internal/diagnostics/log/MoshiJsonFormatterTests.java

```diff
@@ -0,0 +1,46 @@
+package com.microsoft.applicationinsights.agentc.internal.diagnostics.log;
+
+import org.junit.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+public class MoshiJsonFormatterTests {
+
+    private MoshiJsonFormatter formatter;
+
+    @Before
+    public void setup() {
+        formatter = new MoshiJsonFormatter();
+    }
+
+    @After
+    public void tearDown() {
+        formatter = null;
+    }
+
+    @Test
+    public void formatterSerializesSimpleMap() throws Exception {
+        Map<String, Object> m = new HashMap<>();
+        m.put("s1", "v1");
+        m.put("int1", 123);
+        m.put("b", true);
+        assertEquals("{\"b\":true,\"int1\":123,\"s1\":\"v1\"}", formatter.toJsonString(m));
+    }
+
+    @Test
+    public void formatterWithPrettyPrintPrintsPretty() throws Exception {
```

Comment:
Do we need pretty print option?

Reply:
no, but I've been using it for development. The prod config won't have it on.